src/peakrdl_python/templates/peakrdl_python/base.py

```diff
@@ -36,8 +39,28 @@ def inst_name(self) -> str:
         """
         return self.__inst_name
 
+    @property
+    def parent(self) -> Optional[Base]:
+        """
+        parent of the node or field, or None if it has no parent
+        """
+        return self.__parent
+
+    @property
+    def full_name(self) -> str:
```

Comment:
I would favour changing this to `full_inst_name` to make is more consistent with the `inst_name` above

Reply:
I have made this change